tms/src/main/java/com/netcracker/edu/tms/dao/ProjectDaoJPAImpl.java

```diff
@@ -0,0 +1,127 @@
+package com.netcracker.edu.tms.dao;
+
+import com.netcracker.edu.tms.model.Project;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Repository;
+import java.math.BigInteger;
+import java.util.List;
+import javax.persistence.*;
+
+@Repository
+public class ProjectDaoJPAImpl implements ProjectDao {
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
+    public Project getProjectById(BigInteger projectId) {
+        try{
+            LOGGER.info("getEmployeeById called with \""+projectId+"\"");
+            return entityManager.find(Project.class, projectId);
+        }catch (IllegalArgumentException e){
```

Comment:
in which cases we have IllegalArgumentException ?
and it is unchecked exception, that means we does not expect it
moreover we just log it and then throw it again. Who will be the next to face with it?
As for me we don't need to catch it at all